main/reference/zoe-api/zoe-data-types.md

```diff
@@ -83,11 +54,34 @@ is exclusively yours and no one else has access to it.
 
 ## Keyword
 
-An ASCII identifier string that must begin with an upper case letter
-in order to avoid collisions with JavaScript properties such as `toString`.
+A **Keyword** is a unique identifier string within a contract for tying together its
+**proposals**, **payments**, and **payouts**.
+It must be a valid ASCII [identifier](https://developer.mozilla.org/en-US/docs/Glossary/Identifier)
+and start with an upper case letter in order to avoid collisions with JavaScript properties
+such as `toString` when used as a property name in a record.
 (For more detail, see [Why do Zoe keywords have to start with a capital letter? #8241](https://github.com/Agoric/agoric-sdk/discussions/8241).)
 `NaN` and `Infinity` are also not allowed as keywords.
 
+<a name="amountkeywordrecord"></a>
+## KeywordRecord
+
+A **KeywordRecord** is a [CopyRecord](/glossary/#passable) in which every property name
+is a **[Keyword](#keyword)**, such as `harden({ Currency: quatloosBrand, CurrencyUnit: 100n })`.
```

Comment:
That example seems weird. I don't believe anything in our `Record<Keyword, Brand | bigint>`. Did you mean to exhibit a `Record<Keyword, Amount<'nat'>>`? aka a `Record<Keyword, { brand: Brand<'nat'>, value: NatValue }>`

I wonder if it's useful to generalize to KeywordRecord at all. If so, we might as well lean into TypeScript notation: `Record<Keyword, T>`

Reply:
> That example seems weird. I don't believe anything in our `Record<Keyword, Brand | bigint>`. Did you mean to exhibit a `Record<Keyword, Amount<'nat'>>`? aka a `Record<Keyword, { brand: Brand<'nat'>, value: NatValue }>`

No, for that first example I specifically wanted a KeywordRecord that had heterogeneous value types and so was not an AmountKeywordRecord/PaymentKeywordRecord/etc. But if you think this one is confusing, perhaps it could drift even further away from our flavor of DeFi?
```suggestion
is a **[Keyword](#keyword)**, such as `harden({ Remotable: myMap, SupportedMethods: ['delete', 'get', 'has', 'set'] })`.
```

> I wonder if it's useful to generalize to KeywordRecord at all. If so, we might as well lean into TypeScript notation: `Record<Keyword, T>`

I thought about that and actually started with something similar, but ultimately decided not to because a) the site already uses AmountKeywordRecord and PaymentKeywordRecord, and b) we don't explain use of TypeScript syntax and AFAICT don't expect developers to know it. But a thought did occur to me in typing the [last response](https://github.com/Agoric/documentation/pull/876#discussion_r1409666981)—what would you think about using it to help explain the concept for people who are familiar with it?

> ## KeywordRecord
> 
> A **KeywordRecord** is a [CopyRecord](/glossary/#copyrecord) in which every property name
> is a **[Keyword](#keyword)**, such as `harden({ Currency: quatloosBrand, CurrencyUnit: 100n })`.
> Subtypes further constrain property values (for example, an
> **AmountKeywordRecord** is a **KeywordRecord\<Amount>** in which every value is an
> **[Amount](/reference/ertp-api/ertp-data-types.md#amount)** and a
> **PaymentKeywordRecord** is a **KeywordRecord\<Payment>** in which every value is a
> **[Payment](/reference/ertp-api/payment.md)**).

Reply:
> ... for that first example I specifically wanted a KeywordRecord that had heterogeneous value types ...

Such an example is a distraction, no? There is no such data type in the Zoe API.

This is the core of my request for changes.

Reply:
I agree. There are no KeywordRecords with heterogeneous value types, and I don't think we're likely to add them.

Reply:
Okay, updated.

> A **KeywordRecord** is a [CopyRecord](/glossary/#copyrecord) in which every property name
> is a **[Keyword](#keyword)**, such as `harden({ Asset: moolaIssuer, Bid: simoleanIssuer })`.

Reply:
> KeywordRecord that had heterogeneous value types

Please don't. None of our keyword records are heterogeneous, and likely none ever will be, since the point of keywords as names is to support lookup by `[]` user-defined names, and therefore computed property names.